File: app/src/main/java/com/botijasoftware/utils/collision/CollisionSphere.java
package com.botijasoftware.utils.collision;

import com.botijasoftware.utils.Vector3;

public class CollisionSphere extends CollisionVolume {
	private Sphere sphere;
	
	public CollisionSphere(Vector3 center, float radius) {
		sphere = new Sphere(center, radius);
	}
	
	public boolean collides(CollisionVolume c) {
		return c.collides(sphere);
	}
	
	public boolean collides(Sphere s) {
		return sphere.collides(s);
	}
	
	public boolean collides(Vector3 v) {
		return sphere.collides(v);
	}
	
	public boolean collides(Capsule c) {
		return c.collides(sphere);
	}
	
	public boolean collides(Ray r) {
		return r.collides(sphere);
	}
	
	public Vector3 getPosition() {
		return sphere.getCenter();
	}
	
	public void setPosition(Vector3 v) {
		sphere.setCenter(v);
	}
	
	public void setRadius(float r) {
		sphere.setRadius(r);
	}
	
	public float getRadius() {
		return sphere.radius;
	}
	
}

File: app/src/main/java/com/botijasoftware/utils/collision/CollisionSphere.kt
package com.botijasoftware.utils.collision

import com.botijasoftware.utils.Vector3

class CollisionSphere(center: Vector3, radius: Float) : CollisionVolume() {
    private val sphere: Sphere

    init {
        sphere = Sphere(center, radius)
    }

    override fun collides(c: CollisionVolume): Boolean {
        return c.collides(sphere)
    }

    override fun collides(s: Sphere): Boolean {
        return sphere.collides(s)
    }

    override fun collides(v: Vector3): Boolean {
        return sphere.collides(v)
    }

    override fun collides(c: Capsule): Boolean {
        return c.collides(sphere)
    }

    override fun collides(r: Ray): Boolean {
        return r.collides(sphere)
    }

    override var position: Vector3
        get() = sphere.center
        set(v) {
            sphere.center = v
        }

    override var radius: Float
        get() = sphere.radius
        set(r) { sphere.radius = r }

}